cisagov/provision-aws-account
# Support creation of multiple accounts at once

#  #

## 🗣 Description ##

This PR updates the code in this repo so that it can support creation of multiple AWS accounts at once, rather than the previous "one at a time" behavior.
 



## 💭 Motivation and context ##

Eventually, we will need to provision new AWS accounts in bulk- creating accounts one at a time will be too time consuming for us.  Even though Control Tower limits us to a single thread of creating one account at a time, this change allows us to queue up multiple accounts to be created serially.





## 🧪 Testing ##

I was able to do the following:

- Run a single `terraform apply` that successfully created two new Staging accounts.
- Import all existing Staging accounts into my Terraform state (with one exception, but that appears to be due to a bug in the AWS provider), then run a successful `terraform apply` with a tfvars file containing all Staging accounts resulting in no infrastructure changes.






## ✅ Checklist ##






      to reflect the changes in this PR.
